src/map.cpp

```diff
@@ -13,11 +13,15 @@
 
 namespace bpftrace {
 
-int Map::create_map(enum bpf_map_type map_type, const char *name, int key_size, int value_size, int max_entries, int flags) {
+int Map::create_map(enum bpf_map_type map_type, const std::string & name, int key_size, int value_size, int max_entries, int flags) {
+  std::string name_ = name;
+  if (name_[0] == '@') {
```

Comment:
Should check `!fixed_name.empty()` before indexing

Reply:
Done. Although C++11 allows indexing the one-passed-the-end character of a string.